### PR TITLE
Updates WordPressUI.

### LINKED ISSUE #14926: Notifications show incorrect author Gravatars on first load
### Expected behavior
Notifications should show the correct author Gravatar for each notification.

### Actual behavior
When tapping on Notifications and new notifications load, sometimes the new unread notifications in the list have an incorrect author Gravatar on the first load. Pulling down to reload the view will update the notifications to the correct author Gravatar. 

Examples, incorrect Gravatars circled.

Example 1|Example 2|Example 3
---|---|---
CleanShot 2020-09-16 at 20 58 09@2x|CleanShot 2020-09-16 at 20 57 59@2x|CleanShot 2020-09-16 at 20 57 29@2x

### Steps to reproduce the behavior
1. Open WPiOS app
2. Tap on Notifications
3. Let new notifications load
4. Look to see if any of your friends' names don't match their Gravatars
5. Pull down to refresh and the incorrect author Gravatars will update to the correct ones

##### Tested on iPhone X, iOS 13.7, WPiOS 15.3-15.6

### PR DESCRIPTION
**Pull request by diegoreymendez**
Updated WordPressUI to try to resolve #14926.

This should not be merged until a new version of WPUI is released.

Since I wasn't able to reproduce the original issue, I'm not listing this as a fix in the readme file.

I do believe the changes in this PR should address the issues, though.

## Related PRs:

WPUI - https://github.com/wordpress-mobile/WordPressUI-iOS/pull/102

## To test:

1. Check that the "Me" icon looks fine.
2. Go to the notifications tab.
3. Scroll, change subsection, tap on notifications.

Make sure at all times the user icons look fine.

## Regression Notes

1. Potential unintended areas of impact

User images throughout the App.  Gravatar images.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

The are we can impact is what we're trying to fix.  Please check #14926.

I did a full logic review of our code and came up with an updated mechanism to ensure consistency in the images.

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile
Comment (AliSoftware) [inline Podfile]: FYI: I took the liberty to add the `[Status] Blocked` label to be sure we don't accidentally forget to make a new WPUI release and update this reference here before merging

Comment (diegoreymendez) [reply Podfile]: There's a "DO NOT MERGE" label that seems to raise a peril alert.  I'd guess this is more appropriate for the case?

Comment (AliSoftware) [reply Podfile]: Yeah, to be honest those two labels are pretty close to one another semantically, I think they are sometimes used interchangeably. The fact that the "Do Not Merge" label raises a peril alert is a good point and argument for it tho, indeed 👍 

Though, that being said, I also forgot we also [already have a Peril rule which detected when we have a `pod` pointing to a commit](https://github.com/Automattic/peril-settings/blob/master/org/pr/ios-macos.ts#L17-L22) – which was actually triggered before I added my label or yours, according to my email notification history – so I think if we don't set any of those labels (which seems to take precedence over that other rule), Peril will detect it and still ensure we don't accidentally merge a pod pointing to a commit anyway 😅 

Anyway, as long as we make it clear to anyone reviewing the PR that it should not be merged until we update this, that's what's important 🙂

**Review by Gio2018 — APPROVED** [review]
Worked as expected!

**Pull request merged**
